model-optimizer/extensions/ops/elementwise.py

```diff
@@ -131,12 +168,8 @@ def operation(a, b):
 
     @staticmethod
     def type_infer(node):
-        in_type_0 = node.in_port(0).get_data_type()
-        in_type_1 = node.in_port(1).get_data_type()
-        assert in_type_0 == in_type_1, \
-            'Power operation {} has inputs of different data types: {} and {}'.format(
-                node.soft_get('name'), in_type_0, in_type_1)
-        node.out_port(0).set_data_type(in_type_0)
+        override_input_types_of_pow(node)
+        node.out_port(0).set_data_type(node.in_port(0).get_data_type())
```

Comment:
Do we have ONNX layer tests run which proves that the solution is correct?

Reply:
reverted these changes for "problem-3*